File: clients/admin/src/features/employees/types/employee.ts
import type { OrgUnitType } from "@/features/departments/types/org-unit";

/** Mirrors `Organization.Contracts/Employees/EmploymentStatus.cs` — serialized by name. */
export enum EmploymentStatus {
  Active = "Active",
  OnLeave = "OnLeave",
  Terminated = "Terminated",
}

export interface EmployeeMembershipDto {
  orgUnitId: string;
  orgUnitName: string;
  orgUnitType: OrgUnitType;
  levelId?: string | null;
  levelName?: string | null;
  isPrimary: boolean;
  startDate: string;
  endDate?: string | null;
}

export interface EmployeeDto {
  id: string;
  companyId: string;
  userId?: string | null;
  employeeCode: string;
  firstName: string;
  lastName: string;
  dateOfBirth?: string | null;
  gender?: string | null;
  nationalId?: string | null;
  email?: string | null;
  phoneNumber?: string | null;
  address?: string | null;
  hireDate?: string | null;
  terminationDate?: string | null;
  employmentStatus: EmploymentStatus;
  avatarUrl?: string | null;
  memberships: EmployeeMembershipDto[];
}

export interface CreateEmployeeRequest {
  companyId: string;
  employeeCode: string;
  firstName: string;
  lastName: string;
  dateOfBirth?: string;
  gender?: string;
  nationalId?: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  hireDate?: string;
  avatarUrl?: string;
}

export interface EmployeeSearchParams {
  companyId?: string;
  orgUnitId?: string;
  employmentStatus?: EmploymentStatus;
  searchValue?: string;
  pageNumber?: number;
  pageSize?: number;
}

export interface AssignEmployeeOrgUnitRequest {
  orgUnitId: string;
  levelId?: string;
  isPrimary: boolean;
}

export interface UpdateEmployeeMembershipRequest {
  levelId?: string;
  isPrimary: boolean;
}

export interface CreateEmployeeLoginRequest {
  userName: string;
  password?: string;
  email?: string;
  phoneNumber?: string;
}

export interface LinkEmployeeLoginRequest {
  userId: string;
}

File: src/Organization.Api/Domain/Employees/Employee.cs
using StarterKit.Organization.Api.Domain.OrgUnits;
using StarterKit.Shared.Entities;

namespace StarterKit.Organization.Api.Domain.Employees;

public class Employee : AuditableEntity
{
    public string CompanyId { get; set; } = null!;

    public string? UserId { get; set; }

    public string EmployeeCode { get; set; } = null!;

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public DateTimeOffset? DateOfBirth { get; set; }

    public string? Gender { get; set; }

    public string? NationalId { get; set; }

    public string? Email { get; set; }

    public string? PhoneNumber { get; set; }

    public string? Address { get; set; }

    public DateTimeOffset? HireDate { get; set; }

    public DateTimeOffset? TerminationDate { get; set; }

    public EmploymentStatus EmploymentStatus { get; set; } = EmploymentStatus.Active;

    public string? AvatarUrl { get; set; }

    public IList<EmployeeOrgUnitMembership> Memberships { get; set; } = [];
}
